### PR TITLE
Add nats package

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds the NATS package.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## How to test this PR locally

1. Manually start Elastic stack and Nats service 
2. Enable NAts integration from Fleet UI
3. Verify data are populated and both dashboards work.

## Related issues

- Closes https://github.com/elastic/integrations/issues/359

## Screenshots
Screenshot 2020-11-25 at 11 23 09
Screenshot 2020-11-25 at 12 22 57
Screenshot 2020-11-25 at 12 23 09
Screenshot 2020-11-25 at 12 24 15
Screenshot 2020-11-25 at 13 45 32
Screenshot 2020-11-25 at 13 47 47

### CODE REVIEW

**Comment by ChrsMark** [comment]
@elastic/integrations-platforms this one is ready for review!

File: packages/nats/data_stream/connection/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+streams:
+  - input: nats/metrics
+    vars:
+      - name: period
+        type: text
+        title: Period
+        multi: false
+        required: true
+        show_user: true
+        default: 10s
```
Comment (mtojek) [inline packages/nats/data_stream/connection/manifest.yml:13]: nit: not sure if it's not too often. Maybe 30s for all data streams? Is it according to the migration assessment?

File: packages/nats/data_stream/connection/sample_event.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
+{
+    "_index": ".ds-metrics-nats.connection-default-000001",
+    "_type": "_doc",
+    "_id": "llFC_3UBsNbKIce5kE5a",
+    "_version": 1,
+    "_score": null,
+    "_source": {
```
Comment (mtojek) [inline packages/nats/data_stream/connection/sample_event.json:7]: I think in general you can extract the `_source` content from this JSON document.

**Review by mtojek — APPROVED** [review]
LGTM, nice that there're dashboards available ;) Feel free to merge this and iterate in next pull requests if necessary.

File: packages/nats/data_stream/connection/manifest.yml (hunk 1 above)
Comment (ChrsMark) [reply packages/nats/data_stream/connection/manifest.yml:13]: Yeap assessment has it like this: https://github.com/elastic/integrations-dev/blob/master/migration/packages/nats.md#datasets, but we can change it in the future if we see it's not realistic.

**Pull request merged**